x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextField.java

```diff
@@ -0,0 +1,328 @@
+            true,
+            (args, context) -> new SemanticTextField(
+                context.v1(),
+                (List<String>) (args[0] == null ? List.of() : args[0]),
+                (InferenceResult) args[1],
+                context.v2()
+            )
+        );
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<InferenceResult, Void> INFERENCE_RESULT_PARSER = new ConstructingObjectParser<>(
+        INFERENCE_FIELD.getPreferredName(),
+        true,
+        args -> new InferenceResult((String) args[0], (ModelSettings) args[1], (List<Chunk>) args[2])
+    );
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<Chunk, Void> CHUNKS_PARSER = new ConstructingObjectParser<>(
+        CHUNKS_FIELD.getPreferredName(),
+        true,
+        args -> new Chunk((String) args[0], (BytesReference) args[1])
+    );
+
+    private static final ConstructingObjectParser<ModelSettings, Void> MODEL_SETTINGS_PARSER = new ConstructingObjectParser<>(
+        MODEL_SETTINGS_FIELD.getPreferredName(),
+        true,
+        args -> {
+            TaskType taskType = TaskType.fromString((String) args[0]);
+            Integer dimensions = (Integer) args[1];
+            SimilarityMeasure similarity = args[2] == null ? null : SimilarityMeasure.fromString((String) args[2]);
+            return new ModelSettings(taskType, dimensions, similarity);
+        }
+    );
+
+    static {
+        SEMANTIC_TEXT_FIELD_PARSER.declareStringArray(optionalConstructorArg(), TEXT_FIELD);
+        SEMANTIC_TEXT_FIELD_PARSER.declareObject(constructorArg(), (p, c) -> INFERENCE_RESULT_PARSER.parse(p, null), INFERENCE_FIELD);
+
+        INFERENCE_RESULT_PARSER.declareString(constructorArg(), INFERENCE_ID_FIELD);
+        INFERENCE_RESULT_PARSER.declareObject(constructorArg(), (p, c) -> MODEL_SETTINGS_PARSER.parse(p, c), MODEL_SETTINGS_FIELD);
+        INFERENCE_RESULT_PARSER.declareObjectArray(constructorArg(), (p, c) -> CHUNKS_PARSER.parse(p, c), CHUNKS_FIELD);
+
+        CHUNKS_PARSER.declareString(constructorArg(), CHUNKED_TEXT_FIELD);
+        CHUNKS_PARSER.declareField(constructorArg(), (p, c) -> {
+            XContentBuilder b = XContentBuilder.builder(p.contentType().xContent());
+            b.copyCurrentStructure(p);
+            return BytesReference.bytes(b);
+        }, CHUNKED_EMBEDDINGS_FIELD, ObjectParser.ValueType.OBJECT_ARRAY);
+
+        MODEL_SETTINGS_PARSER.declareString(ConstructingObjectParser.constructorArg(), TASK_TYPE_FIELD);
+        MODEL_SETTINGS_PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), DIMENSIONS_FIELD);
+        MODEL_SETTINGS_PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), SIMILARITY_FIELD);
+    }
+
+    /**
+     * Converts the provided {@link ChunkedInferenceServiceResults} into a list of {@link Chunk}.
+     */
+    public static List<Chunk> toSemanticTextFieldChunks(
```

Comment:
This is also used by the `ShardBulkInferenceActionFilter` to translate the response from the inference call.
This class is not in this PR so I am fine if we move it to tests for now but it will need to move back when we introduce the action filter.